src/DataProvider.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+*   DataProvider.cpp
+*   ----------------------
+*   Created on: 2023/09/29
+*   Author: Lankow
+*/
+#include "DataProvider.hpp"
+
+DataProvider::DataProvider()
+    : m_currentTime(DEFAULT_TIME),
+      m_temperature(0),
+      m_roomHumidity(0),
+      m_sensorToWater(NO_SENSOR)
+{
+    for (int i = 0; i < MAX_SENSORS_NO; i++) {
+        m_currentHumidityLvl[i] = ANALOG_PIN_MAX;
+        m_humidityThreshold[i] = ANALOG_PIN_MAX;
+        m_humidityActive[i] = false;
+    }
+}
+
+/* GETTERS */
+
+char* DataProvider::getCurrentTime() {
+    return m_currentTime;
+}
+
+float& DataProvider::getTemperature() {
```

Comment:
Jeśli  nie planujesz modyfikować wartości zwracanej przez funkcje (co swoją drogą jest też turbo złe) to nie ma sensu zwracać tego jako referencje. Żadna oszczedność bo waży tyle samo, a w 64 bitowych systemach wskaznik chyba wazy wiecej (co jest pierdołą ale tylko mowie). Ale najważniejsze że jest to nieczytelne i niebezpieczne